### PR TITLE
Switches to pekko dependencies

### PR DESCRIPTION
**Pull request by nvollmar**

### CODE REVIEW

File: core/src/main/resources/reference.conf [hunk 1]
```diff
@@ -79,10 +79,10 @@ akka.persistence.cassandra {
   journal {
 
     # FQCN of the cassandra journal plugin
-    class = "akka.persistence.cassandra.journal.CassandraJournal"
+    class = "pekko.persistence.cassandra.journal.CassandraJournal"
```
Comment (pjfanning) [inline core/src/main/resources/reference.conf:82]: classes start with org.apache

Comment (nvollmar) [reply core/src/main/resources/reference.conf:82]: thanks for the hint. I just did the package rework. configuration is next.

**Comment by pjfanning** [comment]
most scala 2.12 pekko-connectors jars are now publishing - see https://repository.apache.org/content/groups/snapshots/org/apache/pekko/

should be enough to get the scala 2.12 build of this module working (if you choose a newer snapshot version for pekko-connectors)

**Comment by pjfanning** [comment]
https://github.com/apache/incubator-pekko-connectors/actions/runs/4447589558 passed. It should be possible to get a new connectors snapshot version that supports Scala 2.12 and 2.13

**Comment by pjfanning** [comment]
the doc build issue in latest run (about being unable to find pekko classes) would probably be fixed if you change this in build.sbt.

```
apidocRootPackage := "akka"
```

to

```
apidocRootPackage := "org.apache.pekko"
```

**Comment by nvollmar** [comment]
@pjfanning this would be ready to review

File: build.sbt [hunk 2]
```diff
@@ -19,17 +15,15 @@ lazy val root = project
...
+dumpSchema := (core / Test / runMain).toTask(" org.apache.pekko.persistence.cassandra.PrintCreateStatements").value
 
 lazy val core = project
   .in(file("core"))
   .enablePlugins(Common, AutomateHeaderPlugin, MultiJvmPlugin)
   .dependsOn(cassandraLauncher % Test)
   .settings(
     name := "pekko-persistence-cassandra",
-    libraryDependencies ++= Dependencies.akkaPersistenceCassandraDependencies,
-    Compile / packageBin / packageOptions += Package.ManifestAttributes(
```
Comment (mdedetrich) [inline build.sbt:31]: Why was this removed, or asked differently what is `"Automatic-Module-Name"` for?

File: docs/src/main/paradox/overview.md [hunk 3]
```diff
@@ -14,17 +14,17 @@ This plugin requires **Akka $akka.version$** or later. See [Akka's Binary Compat
   group=com.typesafe.akka
   artifact=akka-persistence-cassandra_$scala.binary.version$
   version=$project.version$
-  symbol=AkkaVersion
+  symbol=pekkoVersion
```
Comment (mdedetrich) [inline docs/src/main/paradox/overview.md:17]: Why did this get changed to lower case, can it be `PekkoVersion` to make sure its consistent with the other projects?  (note there are other cases of this).

File: example/src/main/scala/org/apache/pekko/persistence/cassandra/example/EventProcessorStream.scala [hunk 4]
```diff
@@ -85,7 +85,7 @@ class EventProcessorStream[Event: ClassTag](
   private def readOffset(): Future[Offset] = {
     session
       .selectOne(
-        "SELECT timeUuidOffset FROM akka.offsetStore WHERE eventProcessorId = ? AND tag = ?",
+        "SELECT timeUuidOffset FROM pekko.offsetStore WHERE eventProcessorId = ? AND tag = ?",
```
Comment (mdedetrich) [inline example/src/main/scala/org/apache/pekko/persistence/cassandra/example/EventProcessorStream.scala:88]: Not strictly related to review, but I think we probably need to provide an SQL migration script. Should be discussed on mailing list @pjfanning

**Review by mdedetrich — CHANGES_REQUESTED** [review]
Have a couple of questions but otherwise looking good.

File: build.sbt (hunk 2 above)
Comment (pjfanning) [reply build.sbt:31]: pekko.persistence.cassandra would be a good value

File: project/Dependencies.scala [hunk 5]
```diff
@@ -17,61 +17,61 @@ object Dependencies {
...
+  val pekkoVersionInDocs = pekkoVersion.take(3)
   val cassandraVersionInDocs = "4.0"
-  // Should be sync with the version of the driver in Alpakka Cassandra
+  // Should be sync with the version of the driver in Pekko Connectors Cassandra
   val driverVersionInDocs = "4.6"
 
-  val alpakkaVersion = "3.0.4"
-  val alpakkaVersionInDocs = alpakkaVersion.take(3)
+  val pekkoConnectorsVersion = "0.0.0+54-ee23f1bc-SNAPSHOT"
+  val pekkoConnectorsVersionInDocs = pekkoConnectorsVersion.take(3)
```
Comment (pjfanning) [inline project/Dependencies.scala:27]: could we use `pekkoConnectorsVersionInDocs = "current"` for now? Likewise with pekkoManagementVersion.

File: build.sbt (hunk 2 above)
Comment (nvollmar) [reply build.sbt:31]: I will add that back. If "Automatic-Module-Name" is in the manifest then the module name is taken from it, otherwise the module name is derived from the JAR file name.

File: example/src/main/scala/org/apache/pekko/persistence/cassandra/example/EventProcessorStream.scala (hunk 4 above)
Comment (pjfanning) [reply example/src/main/scala/org/apache/pekko/persistence/cassandra/example/EventProcessorStream.scala:88]: I think we can all agree that replacing `akka` in the schema name makes sense. We should at least log an issue to track what to do about notifying users about the migration SQL they need to run.

File: project/Dependencies.scala (hunk 5 above)
Comment (mdedetrich) [reply project/Dependencies.scala:27]: Missed this, yes we should use `current`

File: example/src/main/scala/org/apache/pekko/persistence/cassandra/example/EventProcessorStream.scala (hunk 4 above)
Comment (mdedetrich) [reply example/src/main/scala/org/apache/pekko/persistence/cassandra/example/EventProcessorStream.scala:88]: Oh definitely we should replace `akka`, but this is as you said as a bare minimum we should at least notify users because it is a breaking change.

Comment (nvollmar) [reply example/src/main/scala/org/apache/pekko/persistence/cassandra/example/EventProcessorStream.scala:88]: Agreed, we will need a good migration guide and tools if possible

File: docs/src/main/paradox/overview.md (hunk 3 above)
Comment (pjfanning) [reply docs/src/main/paradox/overview.md:17]: in that incubator-pekko project, we use PekkoVersion in the `.md` files - eg https://github.com/apache/incubator-pekko/blob/main/docs/src/main/paradox/persistence-fsm.md

File: example/src/main/scala/org/apache/pekko/persistence/cassandra/example/EventProcessorStream.scala (hunk 4 above)
Comment (nvollmar) [reply example/src/main/scala/org/apache/pekko/persistence/cassandra/example/EventProcessorStream.scala:88]: I've created an issue to track this: https://github.com/apache/incubator-pekko-persistence-cassandra/issues/32

File: docs/src/main/paradox/overview.md (hunk 3 above)
Comment (nvollmar) [reply docs/src/main/paradox/overview.md:17]: right, I'll change it to PekkoVersion then

**Comment by nvollmar** [comment]
@mdedetrich could you re-review?
